core/src/main/scala/io/iteratee/internal/Step.scala

```diff
@@ -130,7 +130,7 @@ final object Step { self =>
     onInput: NonEmptyVector[E] => F[Step[F, E, A]],
     onEnd: F[A]
   )(implicit F: Applicative[F]): Step[F, E, A] = new EffectfulCont[F, E, A] {
-    final def feedEl(e: E): F[Step[F, E, A]] = onInput(NonEmptyVector(e))
+    final def feedEl(e: E): F[Step[F, E, A]] = onInput(NonEmptyVector(e, Vector.empty))
```

Comment:
Does Vector.empty allocate at each call? If so maybe better to save an empty and reuse.